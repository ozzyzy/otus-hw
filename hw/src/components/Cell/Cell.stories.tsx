import React from "react";
import {Cell} from "./Cell";
import {number, withKnobs} from "@storybook/addon-knobs";
import {action} from "@storybook/addon-actions";

export default {
    title: "Cell Story",
    decorators: [withKnobs]
}

export const CellStory = () => [
    <Cell
        clickHandler={action("Cell is clicked")}
        x={number("x", 1)}
        y={number("y", 2)}
        key="1-2"
    />
];
