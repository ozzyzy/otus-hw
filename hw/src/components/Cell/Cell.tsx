import React, {FC, useState} from "react";
import styled from "@emotion/styled";

interface ICell {
    key: string;
    x: number;
    y: number;
    clickHandler: (x: number, y: number) => void;
}

const DefaultCell = styled.div`
  display: flex;
  padding: 10px;
  margin: 1px;
  height: 15px;
  width: 15px;
  border: 1px solid black;
`;

export const Cell: FC<ICell> = ({x, y, clickHandler}) => {
    const [content, setContent] = useState('');

    return (
        <DefaultCell
            onClick={() => {setContent(`${x}`); clickHandler(x, y);}}
        > {content}
        </DefaultCell>
    )
}