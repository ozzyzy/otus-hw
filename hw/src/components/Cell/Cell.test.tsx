import React from "react";
import {Cell} from "./Cell";
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";

describe("Cell", () => {
    const x = 1;
    const y = 1;
    const key = '1-1';
    const clickHandler = jest.fn();

    it("should show cell", () => {
        expect(renderer.create(
            <Cell
                x={x}
                y={y}
                key={key}
                clickHandler={clickHandler}/>
            ).toJSON()
        ).toMatchSnapshot();
    });

    it('should click', () => {
        const wrapper = shallow(
            <Cell x={x} y={y} key={key} clickHandler={clickHandler}/>
        );
        wrapper.simulate('click');

        expect(clickHandler).toBeCalledWith(x, y);
    });
});