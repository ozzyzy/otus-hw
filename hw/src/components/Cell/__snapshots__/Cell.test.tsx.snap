// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should show cell 1`] = `
<div
  className="css-cyprvm"
  onClick={[Function]}
>
</div>
`;
